specs/experimental/op-contracts-manager.md

```diff
@@ -153,13 +166,98 @@ This provides the following benefits:
 
 - Contract addresses for a chain can be derived as a function of chain ID without any RPC calls.
 - Chain ID uniqueness is enforced for free, as a deploy using the same chain ID
-will result in attempting to deploy to the same address, which is prohibited by
-the EVM.
+  will result in attempting to deploy to the same address, which is prohibited by
+  the EVM.
   - This property is contingent on the proxy and `AddressManager` code not
-  changing when OP Contracts Manager is upgraded. Both of these are not planned to
-  change.
+    changing when OP Contracts Manager is upgraded. Both of these are not planned to
+    change.
   - The OP Contracts Manager is not responsible for enforcing chain ID uniqueness, so it is acceptable
-  if this property is not preserved in future versions of the OP Contracts Manager.
+    if this property is not preserved in future versions of the OP Contracts Manager.
+
+## Upgrading
+
+### Interface
+
+#### `upgrade`
+
+The `upgrade` method is used by the Upgrade Controller to upgrade the full set of L1 contracts for
+all chains that it controls.
+
+It has the following interface:
+
+```solidity
+function upgrade(ISystemConfig[] _systemConfigs, NewChainConfig[] _newConfigs) public;
```

Comment:
The interface in the design doc is slightly different to this, it has the superchain proxy admin as a parameter: https://github.com/ethereum-optimism/design-docs/blob/main/protocol/l1-upgrades.md#release-process

Reply:
Ahh, yes, that part of the design doc is outdated. It had previously included a proposal to consolidate ALL proxy admin contracts into a single `SuperchainProxyAdmin`. I should have updated the interface
when I removed that proposal but did not. But I think that's OK since the spec will superced the design doc and we don't have a requirement of keeping them current.

Reply:
great callout, I think we'll actually need to provide both the list of SystemConfigs and ProxyAdmins. We can use the superchain-registry to get them and ensure they are matching pairs.

Reply:
```suggestion
function upgrade(ISystemConfig[] _systemConfigs, IProxyAdmins[] _proxyAdmins, NewChainConfig[] _newConfigs) public;
```